app/controllers/rich_text_reactions_controller.rb

```diff
@@ -0,0 +1,37 @@
+class RichTextReactionsController < ApplicationController
+  before_action :build_rich_text_reaction, only: %i[create]
+  before_action :set_rich_text_reaction, only: %i[destroy]
+
+  # POST /rich_text_reactions.turbo_stream
+  def create
+    if @rich_text_reaction.save
+      render :create, status: :created
+    else
+      flash.now[:alert] = 'Reaction could not be created.'
+    end
+  end
+
+  # DELETE /rich_text_reactions/:id.turbo_stream
+  def destroy
+    msg = 'Reaction could not be deleted.'
+    flash.now[:alert] = msg unless @rich_text_reaction.destroy
+  end
+
+  private
+
+  def build_rich_text_reaction
+    @rich_text_reaction = current_user.rich_text_reactions.build(
+      rich_text_reaction_params
+    )
+  end
+
+  def set_rich_text_reaction
+    @rich_text_reaction = RichTextReaction.find(params[:id])
+  rescue ActiveRecord::RecordNotFound
```

Comment:
Is there a specific reason why we are rescuing the error here? Line 17 will also raise an error if the `@rich_text_reaction` can't be destroyed.

Reply:
No specific reason. I just did it out of habit of catching every error I am ever aware of when I used to work with C++ lol.
For `#destroy` on line 17, I have two questions.
1. Are you saying that `ActiveRecord::RecordNotFound` could be raised if I try to destroy a non-existing record? If so, should I also add a rescue there? I’m aware that there are clean ways to do this such as the `#rescue_from` so that I don’t repeat the rescue call.
2. The [API](/Users/wonrhim/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/activerecord-7.0.8/lib/active_record/transactions.rb) says (let me know if this is the wrong method) `#destroy` returns false when a `before_destroy` callback throws an `:abort`. Should this be handled as well?

Reply:
1. If you try to delete a record that does not exist, line 17 will actually raise a `NoMethodError` since `@rich_text_reaction` will be nil. So you would need to add a `rescue` block in the `destroy` method too, unless you integrate the `set_rich_text_reaction` method into `destroy`. I believe this would be the way to go since `set_rich_text_reaction` is only called by `destroy` anyway.
2. As discussed - we will review this later if it becomes an issue